### PR TITLE
Upgrade to Yarn 4

### LINKED ISSUE #8051: Upgrade yarn to v4
https://yarnpkg.com/blog/release/4.0

yarn pnp could be a follow-up.

### PR DESCRIPTION
**Pull request by hotzenklotz**
Updated yarn to version 4 following the [offical migration guide.](https://yarnpkg.com/migration/guide). Except for the `.gitignore`file all files were auto-created/auto-updated.

Other changes:
- Removed some additional `yarn` installation of the `command`-JS-package during the SBT asset compilation. The package is already installed as a dev-dependency of the main yarn WK installation.
- Remove yarn caching from CI. `Restore Cache` (51s) and `Save Cache` (2 min) are slower than just `yarn install` including the download (18s). 

### Steps to test:
1. Make sure that `node` 18+ is installed
2. Run `corepack enable`
3. Run `yarn install`

### Issues:
- fixes #8051

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

File: .gitignore
Comment (hotzenklotz) [inline .gitignore:1]: I slightly grouped some of the more obvious ignored files. Only moved stuff around.

File: .gitignore [hunk 1]
```diff
@@ -42,41 +70,34 @@ node_modules/
...
-temp-webknossos-schema**
-conf/application.conf-e
-.env
-.bloop
-.metals
-metals.sbt
+**/screenshots/*.diff.png
+**/screenshots/*.new.png
+
+# Yarn
```
Comment (hotzenklotz) [inline .gitignore:95]: Added a new `yarn` group as described in their docs: https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored

File: tools/proxy/yarn.lock
Comment (hotzenklotz) [inline tools/proxy/yarn.lock:1]: This file was automatically updated.

**Review by philippotto — COMMENTED** [review]
sweet :man_dancing:

File: .gitignore [hunk 2]
```diff
@@ -42,41 +70,34 @@ node_modules/
...
+
+# Yarn 
+.yarn/*
+!.yarn/cache
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
+tools/proxy/.yarn/install-state.gz
```
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply .gitignore]: Good idea.

**Review by philippotto — APPROVED** [review]
really cool! I'm also very happy about the CI speed up :100:

**Pull request merged**